interface/src/app/utils.js

```diff
@@ -17,4 +17,24 @@ angular.module('adage.utils', [])
 
 // Number of digits in activity value shown on frontend:
 .constant('ActivityDigits', 5)
+
+// Service of global machine learning model, which is shared by multiple
+// modules.
```

Comment:
Creating a new module in `mlmodel` directory for this service will affect many modules on front end. Honestly I don't feel comfortable to do it before I leave in a few hours. If you have a strong preference on it, please add an issue and we can deal with it later. Thanks.